lib/model.js

```diff
@@ -2355,7 +2355,7 @@ class Model {
         return this.QueryInterface.bulkInsert(this.getTableName(options), records, options, attributes).then(results => {
           if (Array.isArray(results)) {
             results.forEach((result, i) => {
-              instances[i].set(this.primaryKeyAttribute, result[this.rawAttributes[this.primaryKeyAttribute].field], {raw: true});
+              instances[i] && instances[i].set(this.primaryKeyAttribute, result[this.rawAttributes[this.primaryKeyAttribute].field], {raw: true});
```

Comment:
use `this.primaryKeyField` instead of `this.rawAttributes[this.primaryKeyAttribute].field`